tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/IR/hlo_ops.cc

```diff
@@ -2253,6 +2290,67 @@ OpFoldResult PadOp::fold(ArrayRef<Attribute> operands) {
   return DenseElementsAttr::get(return_type, result);
 }
 
+//===----------------------------------------------------------------------===//
+// DynamicPadOp
+//===----------------------------------------------------------------------===//
+
+void DynamicPadOp::getCanonicalizationPatterns(
+    OwningRewritePatternList& results, MLIRContext* context) {
+  results.insert<DPadToPad>(context);
+}
+
+static LogicalResult Verify(DynamicPadOp op) {
+  auto input_type = op.operand().getType().dyn_cast<RankedTensorType>();
+  // If operand is unranked, there is very little to verify statically.
+  if (!input_type) {
+    return success();
+  }
+  int input_rank = input_type.getRank();
+
+  auto pad_type = op.padding_value().getType().dyn_cast<RankedTensorType>();
+  if (!pad_type || pad_type.getRank() != 0) {
```

Comment:
You can cast instead of dyn_cast and remove the null check, since you're using HLO_DimensionTensor for these now.

Reply:
done